### PR TITLE
Post Android E2E Test Results on Cancellation and Remove Retry

### PR DESCRIPTION
**Pull request by jekloudaMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description
Post Android E2E test results on cancellation so that we can see test results after timeout. Remove retry from Android E2E tests since the second run may not have enough time to finish before timing out

### Main changes in the PR:

1. Remove retry from Android E2E tests
2. Add on cancelled condition to test results publishing

### Validation performed:

1. Run Android E2E tests, tests should complete and post results on success or failure.

### Unit Tests added:

No unit tests necessary: YAML change

### End-to-end tests added:

No

### Change file added:

No

### Next/remaining steps:

Going to try implementing per-test retries on the Android side for more efficient flakey test mitigation

### CODE REVIEW

**Comment by jekloudaMSFT** [comment]
Successful Android test run with changes: https://office.visualstudio.com/ISS/_build/results?buildId=16623045&view=logs&j=2893a480-1613-5722-9236-ab558bdf2fb7

File: tools/yaml-templates/android-test.yml [hunk 1]
```diff
@@ -52,4 +51,4 @@ steps:
       testRunTitle: 'E2E Tests - Android'
       searchFolder: '$(Agent.BuildDirectory)/androidHost'
       mergeTestResults: true
-    condition: succeededOrFailed()
+    condition: or(succeededOrFailed(), eq(variables['Agent.JobStatus'], 'Canceled'))
```
Comment (KangxuanYe) [inline tools/yaml-templates/android-test.yml:54]: Is the change gonna publish result anyway even if E2E task is cancelled?

Comment (jekloudaMSFT) [reply tools/yaml-templates/android-test.yml:54]: Yes, that's the goal, succeededOrFailed() resolves to false when a task is cancelled due to timeout

**Review by AE-MS — APPROVED** [review]

File: tools/yaml-templates/android-test.yml (hunk 1 above)
Comment (KangxuanYe) [reply tools/yaml-templates/android-test.yml:54]: > Yes, that's the goal, succeededOrFailed() resolves to false when a task is cancelled due to timeout

Great. Another lesson learned as iOS could do the same thing leveraging on what you found. I would do some tests later.

**Review by KangxuanYe — APPROVED** [review]

**Pull request merged**